### PR TITLE
Template functions for recommended kubernetes versions

### PR DESCRIPTION
**Pull request by olemarkus**
This PR implements two template functions for setting recommended k8s version based on info from the given channel.

### CODE REVIEW

File: pkg/util/templater/template_functions.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	funcs["include"] = func(name string, context map[string]interface{}) string {
+		content, err := includeSnippet(tm, name, context)
+		if err != nil {
+			panic(err.Error())
+		}
+
+		return content
+	}
+
+	funcs["ChannelRecommendedKubernetesUpgradeVersion"] = func(version string) string {
```
Comment (rifelpet) [inline pkg/util/templater/template_functions.go:44]: These function names are pretty long.

This is essentially the recommended patch version for a given kubernetes version and the function below is the recommended minor+patch version for a given kops version.

How about:
```
ChannelRecommendedKubernetesPatchVersion
ChannelRecommendedKubernetesMinorVersion
```

or even just
```
ChannelKubernetesPatchVersion
ChannelKubernetesMinorVersion
```
?

Comment (olemarkus) [reply pkg/util/templater/template_functions.go:44]: Not the easiest functions to name :)

I mainly based the function names after the primary underlying function. I don't mind long function names; it boils down to what most intuitive for users.

**Comment by olemarkus** [comment]
I realise that channel is actually something we pass in as a value.
I could pass channel name as a param to these template functions, but I think the overall code would be more awkward.
A better alternative may be to make the value of the channel CLI arg available as a function too.

Any thoughts?

**Comment by rifelpet** [comment]
Yea, I don't have a strong opinion about how these should work or their name though I do see the value in having them. Perhaps your original names are sufficient.

I do think its worthy of some documentation so that users don't have to dig into the source code to find the function names.

I'm also wondering if not specifying --channel should use the default channel or not load any channel and have those functions fail somehow. I suppose it only really impacts airgapped environments since `toolbox template` used to not make any network requests and now it does by default. If we go with the default channel then we should consider a release note mentioning the change in behavior.

**Comment by olemarkus** [comment]
My documentation got lost somewhere ... I had actually written that.

Those who run kops in airgapped envs could point to their local channel files. I would guess they have one in order to update their clusters as well.

I'll dig up those docs again, add a bit of release notes, but otherwise keep this PR as is. There is sufficient time to change anything later on. And I have some more similar functions I want to add such as recommended image.

Recommended image would also require cloud to be passed though. Which means either passing that as a variable or as another CLI arg similar to the channel discussion above.

**Comment by rifelpet** [comment]
Yes, I think all of that sounds good 👍🏻 

Airgapped users can use their local channel file but they'll need to _know_ to add the `--channel` flag pointing to their local file when upgrading to kops 1.20, which is what I think the release notes could help with.

**Comment by olemarkus** [comment]
Actually saving files to disk seems to make a difference ... docs pushed.

**Comment by rifelpet** [comment]
This LGTM. Once the conflict is resolved we can get it merged

**Pull request merged**